apps/frontend/src/app/5_pages/ProposalPage/hooks/useVote.ts

```diff
@@ -0,0 +1,66 @@
+import { useCallback } from 'react';
+
+import { Contract } from 'ethers';
+import { t } from 'i18next';
+
+import { noop } from '@sovryn/ui';
+
+import { TransactionType } from '../../../3_organisms/TransactionStepDialog/TransactionStepDialog.types';
+import { GAS_LIMIT } from '../../../../constants/gasLimits';
+import { useTransactionContext } from '../../../../contexts/TransactionContext';
+import { useAccount } from '../../../../hooks/useAccount';
+import { useGetProtocolContract } from '../../../../hooks/useGetContract';
+import { translations } from '../../../../locales/i18n';
+import { Proposal } from '../../../../utils/graphql/rsk/generated';
+
+const pageTranslations = translations.proposalPage;
+
+export const useVote = (proposal: Proposal, onComplete = noop) => {
+  const { signer } = useAccount();
+  const { setTransactions, setIsOpen, setTitle } = useTransactionContext();
+  const governorAdmin = useGetProtocolContract('governorAdmin');
+
+  const submit = useCallback(
+    async (support: boolean) => {
```

Comment:
Create an enum (e.g. `VoteType` with `Support` and `Reject` values) and use it instead of the boolean variable.